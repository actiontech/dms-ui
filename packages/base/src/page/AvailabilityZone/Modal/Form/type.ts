import { FormInstance } from 'antd';

export type AvailabilityZoneFormType = {
  name: string;
  address: string;
};

export type AvailabilityZoneFormProps = {
  form: FormInstance<AvailabilityZoneFormType>;
  isUpdate?: boolean;
};
