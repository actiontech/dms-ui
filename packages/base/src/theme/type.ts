import { WorkflowRecordStatusEnum } from '@actiontech/shared/lib/api/base/service/common.enum';

export type SideMenuTheme = {
  suffixIconColor: string;
  projectSelector: {
    dropdown: {
      labelColor: string;
      iconColor: string;
      activeIconColor: string;
      groupLabelColor: string;
    };
  };
};

export type GuidanceTheme = {
  padding: number;
  gap: number;
  sceneSegmented: {
    backgroundColor: string;
    borderBottom: string;
  };
  steps: {
    icon: {
      width: number;
      height: number;
      backgroundColor: string;
    };
    connection: {
      width: number;
      gap: number;
      backgroundColor: string;
    };
    step: {
      border: string;
      title: {
        borderRight: string;
        index: {
          color: string;
        };
        text: {
          color: string;
        };
      };
      content: {
        desc: {
          color: string;
        };
      };
    };
  };
  guidanceButton: {
    default: {
      background: string;
      color: string;
    };
    hover: {
      background: string;
      color: string;
    };
  };
};

export type SystemTheme = {
  configButton: {
    backgroundColor: string;
    border: string;
  };
  logo: {
    boxShadow: string;
  };
};

export type DataExportTheme = {
  create: {
    form: {
      baseInfoTitleIconColor: string;
    };
  };
  statistics: {
    auditResultStatusColor: {
      [key in WorkflowRecordStatusEnum]: string;
    };
  };
};

export type IconTheme = {
  bindUser: {
    user: string;
    password: string;
  };
  dataExport: {
    infoCircle: string;
    clock: string;
  };
  home: {
    common: string;
  };
  system: {
    basicTitleTips: string;
    modify: string;
  };
};
