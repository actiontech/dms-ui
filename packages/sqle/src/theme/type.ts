import { WorkflowRecordResV2StatusEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';

export type ExecWorkflowTheme = {
  list: {
    desc: {
      copyIconColor: string;
      hoverCopyIconBackgroundColor: string;
    };
  };
  create: {
    auditResult: {
      auditResultDrawer: {
        numberColor: string;
      };
      toggleButton: {
        borderColor: string;
        color: string;
        activeColor: string;
        bgColor: string;
        activeBgColor: string;
      };
      download: {
        borderColor: string;
        bgColor: string;
        boxShadow: string;
        itemColor: string;
        itemHoverColor: string;
        itemIconColor: string;
      };
    };
    form: {
      baseInfoTitleIconColor: string;
    };
    editForm: {
      titleColor: string;
      projectFlagIconColor: string;
    };
  };
  steps: {
    boxShadow: string;
  };
  common: {
    basicInfo: {
      borderColor: string;
      bgColor: string;
      titleColor: string;
      descColor: string;
    };
    auditResultFilter: {
      borderColor: string;
      bgColor: string;
      auditResultInfo: {
        itemBgColor: string;
        textColor: string;
        schemaValueColor: string;
      };
      options: {
        bgColor: string;
        activeBgColor: string;

        textColor: string;
        textActiveColor: string;
        numColor: string;
        numActiveColor: string;
      };
    };
  };
};

export type ReportStatisticsTheme = {
  bgColor: string;
  loadingColor: string;
  cardShow: {
    titleColor: string;
    iconColor: string;
    numberContColor: string;
    noteColor: string;
    contentFontSize: number;
    contentFontWeight: number;
  };
  CardWrapper: {
    titleColor: string;
    titleBorderColor: string;
  };
  TableTopList: {
    titleColor: string;
    noteTipColor: string;
    colorFillQuaternary: string;
    bgColor: {
      toColor: string;
      line1: string;
      line2: string;
      line3: string;
    };
  };
  ChartContTitle: {
    mainColor: string;
    mainSubColor: string;
    subContColor: string;
  };
  DatabaseTypeOrder: {
    default: {
      fontSize: number;
    };
    tooltip: {
      background: string;
      border: string;
      boxShadow: string;
    };
  };
  DatabaseSourceOrder: {
    default: {
      colorText: string;
      colorTextTertiary: string;
    };
  };
  LicenseStatistics: {
    ChartContTitle: {
      lineColor: string;
    };
    LicenseColumn: {
      defaultColor: string;
      fillColor: {
        column: string;
        columnBackground: string;
        xAxis: string;
        yAxis: string;
        state: {
          active: string;
        };
      };
    };
  };
  WorkOrderState: {
    ChartContTitle: {
      color: string;
      noteSubColor: string;
    };
  };
};

export type StatisticsTheme = {
  rectColor: string[];
  rectColorName: { [key: string]: string };
  auditRateStatus: {
    success: {
      color: string;
      bg: string;
    };
    warning: {
      color: string;
      bg: string;
    };
    error: {
      color: string;
      bg: string;
    };
    tip: {
      color: string;
      bg: string;
    };
  };
  auditResultStatusColor: {
    [key in WorkflowRecordResV2StatusEnum]: string;
  };
};

export type ProjectOverviewTheme = {
  SqlCount: {
    baseColor: string;
    grayColor: string;
  };
  DataSourceCount: {
    health: string;
    risk: string;
  };
  ProjectScore: {
    level: {
      dangerous: string;
      warning: string;
      good: string;
      excellent: string;
    };
    indicator: {
      pointer: string;
      pin: {
        fill: string;
        stroke: string;
      };
    };
  };
  ScanTask: {
    bar: {
      fill: string;
      bg: string;
      activeColor: string;
      label: {
        fill: string;
      };
      toolTip: {
        dotColor: string;
      };
    };
    detail: {
      stroke: string;
    };
  };
  DataSourcePerformance: {
    bar: {
      fill: string;
      bg: string;
    };
    toolTip: {
      dotColor: string;
    };
  };
};

export type WorkflowTemplateTheme = {
  progress: {
    normal: string;
    notice: string;
    warning: string;
    error: string;
    remainColor: string;
  };
  step: {
    honour: string;
    addFile: string;
    audit: string;
    sendPlane: string;
  };
  stepCard: {
    userAvatarBorder: string;
    boxShadow: string;
    hoverBoxShadow: string;
    backgroundColor: string;
    disableBorder: string;
    disabledBackgroundColor: string;
    disableHoverBackgroundColor: string;
    cardDesc: string;
    cardOperator: {
      borderTop: string;
      titleColor: string;
    };
    stepLineColor: string;
    stepReviewIndexTextColor: string;
    closeButtonBoxShadow: string;
    closeButtonFontSize: string;
    cardTitleFontWeight: number;
    descFontSize: string;
    operatorTitleFontWeight: number;
    stepReviewIndexTextFontWeight: number;
    authTextColor: string;
  };
  updateWorkflowTemplateStepInfo: {
    titleFontSize: string;
    titleFontWeight: number;
    infoFontSize: string;
    title: {
      titleWrapperBorderBottom: string;
      titleColor: string;
      titleInfoColor: string;
    };
    info: {
      titleColor: string;
    };
    stepInfoWrapper: {
      border: string;
      backgroundColor: string;
      textColor: string;
    };
    stepNodeAlert: {
      backgroundColor: string;
      alertTitleColor: string;
      alertContentColor: string;
    };
  };
  workflowTemplateAuthInfo: {
    topLevelFontWeight: number;
    authLevelFontWeight: number;
    authLevelFontSize: string;
    authInfoFontSize: string;
    borderBottom: string;
    authInfoColor: string;
  };
};

export type IconTheme = {
  execWorkFlow: {
    minusCircleFilledDisabled: string;
    profileSquareFilled: string;
    databaseFilled: string;
    schemaFilled: string;
    fileList: string;
    clock: string;
  };
  workflowTemplate: {
    common: string;
    userCircleFilled: string;
  };
};

export type DataSourceComparisonTheme = {
  comparisonResultDiffBackgroundColor: string;
  comparisonEntry: {
    card: {
      boxShadow: string;
      borderColor: string;
    };
    timeInfo: {
      color: string;
    };
    overview: {
      card: {
        hoverBoxShadow: string;
        suggestion: {
          color: string;
        };
        mode: {
          same: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          inconsistent: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          baseNotExist: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          comparisonNotExist: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
        };
      };
    };
    treeNode: {
      title: {
        bgColor: string;
      };
      node: {
        inconsistent: {
          color: string;
          baselineBgColor: string;
          comparisonBgColor: string;
        };
        missing: {
          color: string;
          baselineBgColor: string;
          comparisonBgColor: string;
        };
        new: {
          color: string;
          bgColor: string;
        };
        tag: {
          inconsistent: {
            color: string;
            bgColor: string;
          };
          missing: {
            color: string;
            bgColor: string;
          };
          new: {
            color: string;
            bgColor: string;
          };
        };
      };
    };
  };
};

export type KnowledgeTheme = {
  highlight: {
    color: string;
    background: string;
  };
  result: {
    list: {
      boxShadow: string;
      hoverBoxShadow: string;
    };
    description: {
      borderLeftColor: string;
    };
  };
  graph: {
    wrapper: {
      borderColor: string;
      backgroundColor: string;
      dotColor: string;
    };
    container: {
      fullScreenBgColor: string;
    };
    control: {
      borderColor: string;
    };
    edge: {
      color: string;
    };
  };
};
