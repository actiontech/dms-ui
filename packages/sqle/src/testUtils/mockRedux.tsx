import whitelist from '../store/whitelist';
import React from 'react';
import { Provider } from 'react-redux';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';
import sqlManagement from '../store/sqlManagement';
import sqlManagementException from '../store/sqlManagementException';
import pipeline from '../store/pipeline';

const reducers = combineReducers({
  whitelist,
  sqlManagement,
  sqlManagementException,
  pipeline
});

export const storeFactory = (initStore: Dictionary = {}) => {
  return configureStore({
    reducer: reducers,
    preloadedState: initStore
  });
};

export const CustomProvider: React.FC<{
  initStore: Dictionary;
  children: JSX.Element;
}> = (props) => {
  return (
    <Provider store={storeFactory(props.initStore)}>{props.children}</Provider>
  );
};
