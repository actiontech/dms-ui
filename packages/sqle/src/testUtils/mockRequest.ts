import { AxiosResponse } from 'axios';
import ruleTemplate from '@actiontech/shared/lib/api/sqle/service/rule_template';
import configuration from '@actiontech/shared/lib/api/sqle/service/configuration';
import audit_plan from '@actiontech/shared/lib/api/sqle/service/audit_plan';
import { driverMeta } from '../hooks/useDatabaseType/index.test.data';

export const successData = (data: any, otherData?: any) => {
  return {
    code: 0,
    message: '',
    data,
    ...otherData
  };
};

export const failedData = (data?: any, otherData?: any) => {
  return {
    code: 100,
    message: 'error',
    data,
    ...otherData
  };
};

export const resolveImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.resolve({
    status,
    headers,
    config,
    statusText,
    data: successData(data)
  });
};

export const rejectImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.reject({
    status,
    headers,
    config,
    statusText,
    data: failedData(data)
  });
};

export const resolveThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {}
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: successData(data, otherData)
      });
    }, 3000);
  });
};

export const resolveErrorThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {}
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: failedData(data, otherData)
      });
    }, 3000);
  });
};

export const rejectThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((_, rej) => {
    setTimeout(() => {
      rej({
        status,
        headers,
        config,
        statusText,
        data: failedData(data)
      });
    }, 3000);
  });
};

export const mockDriver = () => {
  const spy = jest.spyOn(configuration, 'getDriversV2');
  spy.mockImplementation(() => resolveThreeSecond(driverMeta));
  return spy;
};

export const AuditPlanTypesData = [
  { type: 'default', desc: '自定义', instance_type: '' },
  { type: 'mysql_slow_log', desc: '慢日志', instance_type: 'MySQL' },
  { type: 'mysql_mybatis', desc: 'Mybatis 扫描', instance_type: 'MySQL' },
  { type: 'mysql_schema_meta', desc: '库表元数据', instance_type: 'MySQL' },
  {
    type: 'ali_rds_mysql_slow_log',
    desc: '阿里RDS MySQL慢日志',
    instance_type: 'MySQL'
  },
  {
    type: 'ali_rds_mysql_audit_log',
    desc: '阿里RDS MySQL审计日志',
    instance_type: 'MySQL'
  },
  { type: 'oracle_top_sql', desc: 'Oracle TOP SQL', instance_type: 'Oracle' },
  { type: 'all_app_extract', desc: '应用程序SQL抓取', instance_type: '' },
  { type: 'tidb_audit_log', desc: 'TiDB审计日志', instance_type: 'TiDB' },
  {
    type: 'ocean_base_for_mysql_mybatis',
    desc: 'Mybatis 扫描',
    instance_type: ''
  },
  { type: 'ocean_base_for_mysql_top_sql', desc: 'Top SQL', instance_type: '' }
];
export const mockUseAuditPlanTypes = () => {
  const spy = jest.spyOn(audit_plan, 'getAuditPlanTypesV1');
  spy.mockImplementation(() => resolveThreeSecond(AuditPlanTypesData));
  return spy;
};

export const mockUseRuleType = () => {
  const spy = jest.spyOn(ruleTemplate, 'getRuleTypeByDBTypeV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([
      {
        rule_count: 0,
        rule_type: 'DML1'
      },
      {
        rule_count: 3,
        rule_type: 'DML2'
      }
    ])
  );

  return spy;
};
