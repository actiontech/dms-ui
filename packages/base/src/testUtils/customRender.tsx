import { Dictionary } from '@actiontech/shared/lib/types/common.type';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/system';
import { Provider } from 'react-redux';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import { storeFactory } from '@actiontech/shared/lib/testUtil/mockRedux';
import lightTheme from '@actiontech/shared/lib/theme/light';
import basePackageTheme from '../theme/light';

export type RenderParams = Parameters<typeof render>;

const themeData = {
  ...lightTheme,
  ...basePackageTheme
};

export const renderWithReduxAndTheme = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={themeData}>
        <Provider store={storeFactory(initStore)}>{children}</Provider>
      </ThemeProvider>
    ),
    ...option
  });
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
