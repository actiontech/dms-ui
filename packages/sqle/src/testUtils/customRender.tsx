import {
  RenderResult,
  render,
  renderHook as renderHookByReact
} from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
  useLocation
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { storeFactory } from './mockRedux';
import { ConfigProvider, theme as antdTheme } from 'antd';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { mount, shallow } from 'enzyme';
import zhCN from 'antd/locale/zh_CN';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';

import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRouterAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <BrowserRouter>
      <Provider store={storeFactory(initStore)}>{ui}</Provider>
    </BrowserRouter>,
    option
  );
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState)
    }
  } as any);
};

export const renderHooksWithTheme = <TProps, TResult>(
  hooks: (props: TProps) => TResult
) => {
  return renderHookByReact(hooks, {
    wrapper: ({ children }) => (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm }}
      >
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
        </StyledEngineProvider>
      </ConfigProvider>
    )
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
    </StyledEngineProvider>,
    option
  );
};

export const renderWithThemeAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Provider>,
    option
  );
};

export const renderWithThemeAndServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(
    <Router {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Router>,
    option
  );
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <StyledEngineProvider injectFirst>
              <ThemeProvider theme={themeData}>{children}</ThemeProvider>
            </StyledEngineProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: themeData
    }
  });
};

export const shallowWithRouter = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: MemoryRouter
  });
};

export const renderLocationDisplay = (): [
  () => RenderResult,
  () => JSX.Element
] => {
  const LocationComponent = () => {
    const location = useLocation();

    return <div data-testid="location-display">{location.pathname}</div>;
  };

  return [() => renderWithRouter(<LocationComponent />), LocationComponent];
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
