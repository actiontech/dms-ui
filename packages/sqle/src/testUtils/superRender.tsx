import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';
import { renderHook } from '@testing-library/react-hooks';
import { storeFactory } from './mockRedux';

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const sqleSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData,
    storeFactory: storeFactory
  });
};

export const sqleSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData,
    storeFactory: storeFactory
  });
};
