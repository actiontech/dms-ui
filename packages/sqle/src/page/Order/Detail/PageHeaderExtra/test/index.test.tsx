import { OrderDetailPageHeaderExtraProps } from '../index.type';
import OrderDetailPageHeaderExtra from '..';

import { cleanup, fireEvent, act } from '@testing-library/react';
import { renderWithTheme } from '../../../../../testUtils/customRender';
import { mockUseCurrentUser } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentUser';

describe('sqle/Order/Detail/OrderDetailPageHeaderExtra', () => {
  const customRender = (params: OrderDetailPageHeaderExtraProps) => {
    return renderWithTheme(
      <OrderDetailPageHeaderExtra {...params} />
    );
  };

  beforeEach(() => {
    jest.useFakeTimers();
    mockUseCurrentUser();
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
    cleanup();
  });

  it('render snap', () => {
    const { baseElement } = customRender({
      projectName: 'project name'
    });
  })
});
