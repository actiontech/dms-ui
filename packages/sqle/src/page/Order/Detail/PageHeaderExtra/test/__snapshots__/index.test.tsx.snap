// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqle/Order/Detail/OrderDetailPageHeaderExtra render snap 1`] = `
<body>
  <div>
    <div
      class="css-eocfs5"
    >
      <div
        class="close-order-button-wrapper"
        hidden=""
      >
        <button
          class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-dangerous basic-button-wrapper css-geipcv"
          type="button"
        >
          <span>
            关闭工单
          </span>
        </button>
        <div
          class="ant-divider css-dev-only-do-not-override-txh9fw ant-divider-vertical order-detail-page-header-divider"
          role="separator"
        />
      </div>
      <button
        class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
        hidden=""
        type="button"
      >
        <span>
          全部驳回
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
        hidden=""
        type="button"
      >
        <span>
          审核通过
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
        hidden=""
        type="button"
      >
        <span>
          批量立即上线
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
        hidden=""
        type="button"
      >
        <span>
          标记为人工上线
        </span>
      </button>
      <div
        class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        hidden=""
      >
        <div
          class="ant-space-item"
          style="margin-right: 0px;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-dangerous basic-button-wrapper css-geipcv"
            type="button"
          >
            <span>
              中止上线
            </span>
          </button>
        </div>
        <div
          class="ant-space-item"
          style="margin-right: 0px;"
        >
          <div
            class="ant-divider css-dev-only-do-not-override-txh9fw ant-divider-vertical order-detail-page-header-divider"
            role="separator"
          />
        </div>
        <div
          class="ant-space-item"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
            type="button"
          >
            <span>
              刷新工单
            </span>
          </button>
        </div>
      </div>
      <div
        class="toggle-order-detail-wrapper"
        hidden=""
      >
        工单详情
      </div>
    </div>
  </div>
</body>
`;

exports[`sqle/Order/Detail/OrderDetailPageHeaderExtra render snap when btn is hidden 1`] = `
<body>
  <div>
    <div
      class="css-eocfs5"
    >
      <div
        class="close-order-button-wrapper"
        hidden=""
      >
        <button
          class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-dangerous basic-button-wrapper css-geipcv"
          type="button"
        >
          <span>
            关闭工单
          </span>
        </button>
        <div
          class="ant-divider css-dev-only-do-not-override-txh9fw ant-divider-vertical order-detail-page-header-divider"
          role="separator"
        />
      </div>
      <button
        class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
        hidden=""
        type="button"
      >
        <span>
          全部驳回
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
        hidden=""
        type="button"
      >
        <span>
          审核通过
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
        hidden=""
        type="button"
      >
        <span>
          批量立即上线
        </span>
      </button>
      <button
        class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-primary basic-button-wrapper css-geipcv"
        hidden=""
        type="button"
      >
        <span>
          标记为人工上线
        </span>
      </button>
      <div
        class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        hidden=""
      >
        <div
          class="ant-space-item"
          style="margin-right: 0px;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-dangerous basic-button-wrapper css-geipcv"
            type="button"
          >
            <span>
              中止上线
            </span>
          </button>
        </div>
        <div
          class="ant-space-item"
          style="margin-right: 0px;"
        >
          <div
            class="ant-divider css-dev-only-do-not-override-txh9fw ant-divider-vertical order-detail-page-header-divider"
            role="separator"
          />
        </div>
        <div
          class="ant-space-item"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
            type="button"
          >
            <span>
              刷新工单
            </span>
          </button>
        </div>
      </div>
      <div
        class="toggle-order-detail-wrapper"
        hidden=""
      >
        工单详情
      </div>
    </div>
  </div>
</body>
`;

exports[`sqle/Order/Detail/OrderDetailPageHeaderExtra render snap when btn is hidden 2`] = `
<body>
  <div>
    <div
      class="css-eocfs5"
    >
      <div
        class="close-order-button-wrapper"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default ant-btn-dangerous basic-button-wrapper css-geipcv"
          type="button"
        >
          <span>
            关闭工单
          </span>
        </button>
        <div
          class="ant-divider css-dev-only-do-not-override-txh9fw ant-divider-vertical order-detail-page-header-divider"
          role="separator"
        />
      </div>
      <div
        class="ant-space css-dev-only-do-not-override-txh9fw ant-space-horizontal ant-space-align-center"
        hidden=""
      >
        <div
          class="ant-space-item"
          style="margin-right: 0px;"
        />
        <div
          class="ant-space-item"
          style="margin-right: 0px;"
        >
          <div
            class="ant-divider css-dev-only-do-not-override-txh9fw ant-divider-vertical order-detail-page-header-divider"
            role="separator"
          />
        </div>
        <div
          class="ant-space-item"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-674gwq ant-btn-default basic-button-wrapper css-geipcv"
            type="button"
          >
            <span>
              刷新工单
            </span>
          </button>
        </div>
      </div>
      <div
        class="toggle-order-detail-wrapper"
        hidden=""
      >
        工单详情
      </div>
    </div>
  </div>
</body>
`;
